### PR TITLE
UEFI fixes and comments addressed

### PR DESCRIPTION
**Pull request by DaanHoogland**
## Description

This is an attempt to address vmware issues with  #3638, to prevent having to revert the entire PR. see #3978 for the issue.

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

### CODE REVIEW

**Comment by DaanHoogland** [comment]
@blueorangutan package

File: engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java [hunk 1]
```diff
@@ -1315,6 +1312,26 @@ public void orchestrateStart(final String vmUuid, final Map<VirtualMachineProfil
         }
     }
 
+    private void logUefiParameters(Map<VirtualMachineProfile.Param, Object> params) {
```
Comment (weizhouapache) [inline engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java]: @DaanHoogland is it possible params = null ?

Comment (DaanHoogland) [reply engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java]: /me checking

Comment (DaanHoogland) [reply engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java]: good spot, tnx

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan help

**Comment by DaanHoogland** [comment]
@blueorangutan test centos7 vmware-67u3

**Comment by DaanHoogland** [comment]
@blueorangutan test centos7 vmware-67u3

**Comment by DaanHoogland** [comment]
@pavanaravapalli I saw #3985. happy to go with that, but please consider the changes here as well as they address some of the comments made. Let me know and I'll close this.

**Comment by DaanHoogland** [comment]
similar code merged in #3985,
remaining will be added to #4021

**Pull request closed without merging**